.github/workflows/beam_PreCommit_Java.yml

```diff
@@ -172,10 +173,11 @@ jobs:
             -PdisableSpotlessCheck=true \
             -PdisableCheckStyle=true \
       - name: Upload test report
+        if: failure()
```

Comment:
Should this be `if: always()`?

Reply:
There will be a third party workflow generating junit reports: #28075 to be added to all tests. This is available as a complement if one ones to see raw reports so failure() should suffice. I did not investigate how big the artifact could be so I would suggest roll out to failure() first

Reply:
Ok, SGTM